### ISSUE TITLE
Do the transformations apply to BOTH the image AND its instance segmentation mask?  Or do the transformations only apply to the image and not to the segmentation mask?

### ISSUE THREAD

**Issue by cpoptic** [albu/albumentations#444]
## 🐛 Bug

Not necessarily a bug, but a very important point to clarify with the albumentation developers regarding preservation of coordinates of bounding_boxes/segmentation_masks of an image when applying image transformations to the image.  

For example, the goal is to clarify and ensure that a 10 pixel vertical translation of the image also produces along with it a 10 pixel vertical translation in the bounding boxes and segmentation masks.

Or a second example: assert that the bounding boxes and segmentation masks both rotate with the image when applying a rotational transformation.

Do the transformations in Albumentations library apply to BOTH the image as well as the mask for instance segmentation task?

Or do the transformations only apply to the image and not to the segmentation mask?

It is absolutely crucial to confirm that both the bounding_boxes/segmentation_masks and their associated image *both* transform such that the boxes/masks coordinates are maintained in the new transformed coordinate system.

Because if this does not hold, the RPN_loss, BBOX_loss, and mask_losses will be adversely affected, since the new (augmented/transformed) locations of the bboxes/masks will no longer match.

This bug/question was motivated while training an instance segmentation model on Matterport's implementation of Mask RCNN.

I began noticing that when I add a few kinds of augmentations, the training loss suffers greatly compared to when I run the exact same model but *without* the augmentations.  Augmentations should not hurt the training loss by such a great magnitude, so this behavior puzzled me.

Could you please confirm that the Albumentation transformations preserve the segmentation mask polygon coordinates when the image is transformed (translated, scaled, rotated, etc)?

My theory is that the image transformations may be changing the image but not the segmentation masks, thus causing the model validation accuracy to decrease rather than increase, which is what we should expect after applying image augmentations, right?

Thanks for your help.

**Comment by BloodAxe** [comment]
Sure thing, images, masks, boxes and key points are transformed in a consistent way.
